FEAT: Adds support for updating HCP Packer registry channels

tanmay-hc opened the pull request:
Enables specifying channel IDs in the build configuration to automatically update existing HCP Packer registry channels to point to the new build version upon completion. Improves workflow automation and reduces manual channel management for users.

mogrogan reviewed (commented):
Would it be possible to add some tests?

tanmay-hc committed:
FEAT: Adds support for updating HCP Packer registry channels

Enables specifying channels in the build configuration to automatically update existing HCP Packer registry channels to point to the new build version upon completion. Improves workflow automation and reduces manual channel management for users.

tanmay-hc committed:
Added Test Cases for Channel Assignment

devashish-patel commented on internal/hcp/registry/types.bucket.go:
```diff
@@ -244,6 +246,30 @@ func (bucket *Bucket) uploadSbom(ctx context.Context, buildName string, sbom pac
+	}
+
+	for _, channel := range bucket.Channels {
+		log.Printf("[INFO] Updating channel %s to version %s", channel, bucket.Version.ID)
```
`Assigning version XYZ to ABC channel`

devashish-patel reviewed (dismissed):
Overall LGTM. Just minor verbiage changes required before merging. Also, please add HPATs. 

A question -  Currently channel assignment logs are visible only with `PACKER_LOG=1`. Is it possible to show it as a top level regular logs? 

FYI, I used following template for testing:

```hcl
source "docker" "ubuntu" {
  image  = "ubuntu:20.04"
  commit = true
}

hcp_packer_registry {
  bucket_name = "test-channel-assignments"
  channels = ["stable", "beta"]
}

build {
  sources = ["source.docker.ubuntu"]
}
```

tanmay-hc committed:
Adding `ui` object support in Bucket for Logs

devashish-patel committed:
Verbiage change
